app/main/routes.py

```diff
@@ -256,6 +256,10 @@ def series_view(series_id):
     user_id = current_user.id
     # Retrieve the Series that was selected
     series = Series.query.filter_by(id=series_id).first_or_404()
+    study_id = series.study_id
+    device_id=series.device_id
+    study = Study.query.filter_by(id=study_id).one_or_none()
+    device = Device.query.filter_by(id=device_id).first()
```

Comment:
why did you choose different functions for these?
above there is use of `first_or_404` and then `one_or_none` and `first`. I'm wondering what the difference is between these, and what's the rationale for using one over the other.